src/PokemonGeneral.java

```diff
@@ -0,0 +1,58 @@
+abstract class PokemonGeneral {
+    private String name;
+    private String type;
+    private int hp;
+    private int level;
+
+    public PokemonGeneral(String name, int hp, int level) {
+        this.name = name;
+        this.hp = hp;
+        this.level = level;
+    }
+
+    abstract void sound();
+
+    public void tackle(String name) {
+
+        System.out.println(name + " doet je nu tackelen");
```

Comment:
Je haalt hier de naam uit de parameter, terwijl je bijvoorbeel `pikachu.tackle()` aanroept. Dan zou ik verwachten dat pikachu je tackled, maar als ik zeg `pikachu.tackle("Johan")` dan tackled Johan mij, terwijl ik het aan pikachu vraag. 

Beter kun je hier de naam uit de klasse halen met `this.name`. 

De name uit de parameter kun je nog wel gebruiken als bijvoorbeeld: 
`this.name + " valt " + name + " aan met tackle"`
Dit word dat "pikachu van johan aan met tackle"